crates/heimdall/proto/heimdall.proto

```diff
@@ -0,0 +1,37 @@
+syntax = "proto3";
+
+import "google/protobuf/empty.proto";
+
+package dev.restate.heimdall;
+
+service Heimdall {
+  // Register node at controller
+  rpc RegisterNode(RegistrationRequest) returns (RegistrationResponse);
+
+  // Heartbeat the controller
+  rpc Heartbeat(HeartbeatRequest) returns (HeartbeatResponse);
+}
+
+message RegistrationRequest {
+  uint32 node_id = 1;
+}
+
+message RegistrationResponse {
+  message Acceptance {}
+  message Rejection {
+    string reason = 1;
+  }
+
+  oneof result {
+    Acceptance acceptance = 1;
+    Rejection rejection = 2;
+  }
+}
+
+message HeartbeatRequest {
+  uint32 node_id = 1;
```

Comment:
Nit: Can we use typed messages for ID types? i.e. `NodeID` (this can be in a common proto file since I'm pretty sure we will use it in various places)

Reply:
👍 will change it.